src/hidet/runtime/cuda/cudnn.cpp

```diff
@@ -0,0 +1,616 @@
+                CUDNN_ATTR_OPERATION_CONVOLUTION_FORWARD_CONV_DESC,
+                CUDNN_TYPE_BACKEND_DESCRIPTOR, 1, &cDesc));
+    CHECK_CUDNN(cudnnBackendSetAttribute(fprop, CUDNN_ATTR_OPERATION_CONVOLUTION_FORWARD_ALPHA,
+                get_attribute_type_from_compute_type(compType), 1, p_alpha));
+    CHECK_CUDNN(cudnnBackendSetAttribute(fprop, CUDNN_ATTR_OPERATION_CONVOLUTION_FORWARD_BETA,
+                get_attribute_type_from_compute_type(compType), 1, p_beta));
+    CHECK_CUDNN(cudnnBackendFinalize(fprop));
+
+    // Build the operation graph descriptor
+    cudnnBackendDescriptor_t op_graph;
+    CHECK_CUDNN(cudnnBackendCreateDescriptor(CUDNN_BACKEND_OPERATIONGRAPH_DESCRIPTOR, &op_graph));
+    CHECK_CUDNN(cudnnBackendSetAttribute(op_graph, CUDNN_ATTR_OPERATIONGRAPH_OPS,
+                CUDNN_TYPE_BACKEND_DESCRIPTOR, 1, &fprop));
+    CHECK_CUDNN(cudnnBackendSetAttribute(op_graph, CUDNN_ATTR_OPERATIONGRAPH_HANDLE,
+                CUDNN_TYPE_HANDLE, 1, &cur_handle));
+    CHECK_CUDNN(cudnnBackendFinalize(op_graph));
+
+    // Set up engine config
+    cudnnBackendDescriptor_t engine;
+    CHECK_CUDNN(cudnnBackendCreateDescriptor(CUDNN_BACKEND_ENGINE_DESCRIPTOR, &engine));
+    CHECK_CUDNN(cudnnBackendSetAttribute(engine, CUDNN_ATTR_ENGINE_OPERATION_GRAPH,
+                CUDNN_TYPE_BACKEND_DESCRIPTOR, 1, &op_graph));
+    // TODO: Is it okay to hardcode the engine to be CUDNN_ATTR_ENGINE_GLOBAL_INDEX 0?
+    // As mentioned here: https://docs.nvidia.com/deeplearning/cudnn/developer/graph-api.html,
+    // Engine selection should be determined based on some heuristics.
+    int64_t gidx = 0;
+    CHECK_CUDNN(cudnnBackendSetAttribute(engine, CUDNN_ATTR_ENGINE_GLOBAL_INDEX,
+                CUDNN_TYPE_INT64, 1, &gidx));
+    CHECK_CUDNN(cudnnBackendFinalize(engine));
+
+    cudnnBackendDescriptor_t engcfg;
+    CHECK_CUDNN(cudnnBackendCreateDescriptor(CUDNN_BACKEND_ENGINECFG_DESCRIPTOR, &engcfg));
+    CHECK_CUDNN(cudnnBackendSetAttribute(engcfg, CUDNN_ATTR_ENGINECFG_ENGINE,
+                CUDNN_TYPE_BACKEND_DESCRIPTOR, 1, &engine));
+    CHECK_CUDNN(cudnnBackendFinalize(engcfg));
+
+    // Set up the execution plan
+    cudnnBackendDescriptor_t plan;
+    CHECK_CUDNN(cudnnBackendCreateDescriptor(CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR, &plan));
+    CHECK_CUDNN(cudnnBackendSetAttribute(plan, CUDNN_ATTR_EXECUTION_PLAN_HANDLE, CUDNN_TYPE_HANDLE, 1, &cur_handle));
+    CHECK_CUDNN(cudnnBackendSetAttribute(plan, CUDNN_ATTR_EXECUTION_PLAN_ENGINE_CONFIG,
+                CUDNN_TYPE_BACKEND_DESCRIPTOR, 1, &engcfg));
+    CHECK_CUDNN(cudnnBackendFinalize(plan));
+
+    int64_t workspaceSize;
+    CHECK_CUDNN(cudnnBackendGetAttribute(plan, CUDNN_ATTR_EXECUTION_PLAN_WORKSPACE_SIZE,
+                CUDNN_TYPE_INT64, 1, NULL, &workspaceSize));
+    
+    void *dev_ptrs[3] = {ptr_x, ptr_w, ptr_y}; // device pointers
+    int64_t uids[3] = {'x', 'w', 'y'};
+    void *workspace = hidet_cuda_malloc_async(workspaceSize, cur_stream);
```

Comment:
It might be better to use the workspace shared by all hidet operators (i.e., https://github.com/hidet-org/hidet/blob/main/include/hidet/runtime/cuda/context.h#L46). 

When we run the operator in the second time, there will not be any memory allocation. Thus, it can also be used in cudaGraph.